docs/tutorials/02a_training_a_quantum_model_on_a_real_dataset.ipynb

```diff
@@ -0,0 +1,891 @@
oeq+j3us6fH0RCQ1KoEVE5JgxMSuZVbmlHUpiF23Zx81PfsyErGTuvXoykZ4wxg9MJjYyPOAyjq89sZwvP7as3dcWkdCiBFpERI4ZE7NSOFBdx5bCA+163JrdpXzp0aUMTo3loS9OIzbSN4c+IjyM6TmpLNxS1OY5SipreGt9Pu9syGe/Sj5EejUl0CIicsw4NJFwZ0mbx3q9juU7i/n96xv4woNLSIqJ4NHrp5Mce3jLullD+7ClsIL8sqpWz/f2+gLqvA6vg7fW53f0KYhICFAbOxEROWYMSYsjIdrDx7tK+Oy0rCb7D9bU8/7mIt5al8/bGwooOlBNeJgxIyeVX100jsykmCaPmTU0DYCFW4r4zKSBLV779bV5ZCRGEx5mvLE2j89ObXp9EekdlECLiMgxIyzMmJiVzIdb9/HMslz2lBxkT8lBdvv/3FV8kJo6LwlRHk4amc7pY/oxd0RfkmIjWjzn6MxEkmIiWLh5X4sJ9MGaehZ8UsjlU7MICzOe+HAnFdV1xEXpv1GR3kiffBEROaZMHZzKe59s4ranVwK+Zb4HJEczol8Cp47ux4nD05mek0qkJ7AqxfAwY+aQPizcsg/nHGZN+1XP31RIVa2XM8dmEBZmPPTBduZtLOTc8e1fllxEjn1BS6DN7EHgPKDAOTeumf1XAd8DDCgHvuKcWxmseEREJDR8+aQhzBzah74JUWQkRXd4afDGZg3rw2tr89i1/yCD+sQ22f/G2jySYyOYnpOKmdEnLpLX1+YpgRbppYI5ifBh4KxW9m8DTnLOHQf8Erg/iLGIiEiIiI4IZ3pOKtlpcV2SPINvIiHQbDeO2novb63P59RR/fCEhxEeZpw2uh/vbiiguq6+S67f2320fT+vrt7b3WGIBCxoCbRzbgGwv5X9C51zxf67i4GWZ26IiIgE0dD0eNIToprtB7146z7Kquo4c2y/Q9vOHNeP8uo6LQPeBZxz/ODZ1Xz3v6u0SI0cM3pKG7vrgVdb2mlmN5rZUjNbWlhYeBTDEhGR3sDMmDX00zroxl5fm0dMRDgnjkg/tG3W0DTiIsN5Y21el8axbk8Zs+94hzte3dBrRrdX7Cphc8EByqvq2JhX3t3hiASk2xNoMzsZXwL9vZaOcc7d75yb6pybmp6e3tJhIiIiHTZraB+KDlSzueDTRVq8Xscba/M5aUT6YeUi0RHhnDyqL2+uy++yUdPSg7V85YllFFfWcO/8LVzwlw9Ys7u0S87dkz2zLJeIcN/EzSXbNKIvx4ZuTaDNbDzwT+BC55w+NSIi0m0+7Qf96X9HK3JLKCiv5qxxGU2OP3NsBkUHali2o7jJvvbyeh23PrWC3cUHeez66Tz4xansr6zhor9+wN1vf0JdvbfT1+iJqmrreXHlHs4b358ByTEs2d5i5adIj9JtCbSZDQKeBT7vnNvUXXGIiIgAZKXGMjAlhg82fzqR8PW1eXjCjJNH9W1y/NyR6USGh/F6F5Rx/G3eZt5aX8CPzx3NlMGpnDKqH29880TOOS6TP765iUv+vpDNBaFX3vDGunzKq+q4bMpApueksmTb/iYlNCI9UdASaDN7ElgEjDSzXDO73sxuMrOb/If8FOgD/M3MVpjZ0mDFIiIiEojZQ9NYvHUf9V6Hc77yjZlD+5AU03QhloToCOYMT+P1tXmdSvoWbCrkD29u4sKJ/blmVvah7Slxkdx9xST+euVkdu6v5Jy73+etdaG1hPjTS3cxIDmG44f0YXpOKkUHathWVNHdYYm0KZhdOK5wzmU65yKccwOdcw845+51zt3r33+Dcy7FOTfRf5sarFhEREQCMWtYH8qq6li3p4xPCg6wraiCM8c2Ld9ocObYfuQWH2TtnrIOXS+3uJJb/v0xI/om8NuLj2t2EZdzx2fy+rdOZEhaHD95YQ0Ha0JjcuHe0oO8v7mIS6YMJCzMmJ6TCsCSbSrjkJ6v2ycRioiI9BQzh3zaD/q1NXmYwRlj+rV4/Gmj+xFmdKgbR1VtPV99Yjl19Y6/Xz2Z2MiW1zbrmxDNzy4Yy97SKh54f2tA5//Zi2v5ztM9d32yZ5fvxjm4dLKvi+2QtDjS4iOVQMsxQQm0iIiIX9/EaIb1jWfhln28vjaPSVnJ9E2MbvH4PvFRTMtO5fW17S+t+Pn/1rEqt5S7PjuBIenxbR5//JA+nDm2H3+bt4WC8qpWj3151V4eXridp5fl9shOHs45nlmWy4yc1EMrP5r5RqE/VAItxwAl0CIiIo3MGtqHRVv2sXZPWavlGw3OHJvBxvzydtXuPvTBNp5cspOvzB0a0DUafP/s0dTWe/njGy3PvS8sr+bHz69m3IBE4qM83LcgsBHro2nZjmK2FVVw6ZTD11Cbnp3K7pKD5BZXdlNkIoFRAi0iItLIrKF9qPG3jQskuT3Dv0JhoN04nl2ey8//t44zxvTj1tNHtCu2nLQ4Pn98Nv9Zuot1zdRdO+f40XOrqaip50+XT+TKGYN4edUedu0PLCFt72TIhVuK2JRfjredvbCfWZZLbGQ45xyXedj26Tm+EpqP1M5Oejgl0CIiIo3MyOmDGYzKSCA7La7N4wemxDJuQCLPLd9NSWVNq8e+uS6f7zyzillD+3D3FZPwhLf/v+FvnDqMxOgIfvPK+iYJ7/MrdvPGunxuO2MEw/omcO3sbMLMeOD9bW2e99nluZxw57sBJ9vvbMjnyn98yBn/t4Apv3qTGx9dygPvb2PN7tJWF5eprKnjpVV7Oee4TOKiDq/7HpmRQGK0R3XQ0uMpgRYREWkkJS6SG+bk8NWThwX8mJtOGsqWwgOc+acFzN9U2Owxi7bs42v/Ws64/onc/4Wph61s2B7JsZHccupw3t9cxLyNn14rr7SK219Yy9TBKVw/ZwgAmUkxXDhxAP/5aBfFFS0n9w2PzS0+yI+fX9PmSHRVbT23v7iWYX3jufPS8Zw6uh8b8sr55UvrOO8v7zPx52/w8/+tpbSytsljX1+bx4FqX+/
```

Comment:
> Should we be clear that we are just doing a 4 to 2 to show the technique, as in its not about that we can't handle it just that say to make things smaller/faster we can also choose to do this?

Agree. Added: `This dimensionality reduction is for educational purposes only. As you could see in the previous section we can train a quantum model using all 4 features in the dataset.`

> 
> > We'll move at a faster pace as the steps are the same except the first one where we apply a PCA transformation
> 
> Basically we are to do the same classical and quantum classification procedure as before/above but now with this transformed data.

That's right.

> 
> Are the same concerns/considerations for PCA the same for the dataset when used for quantum and for classical - say given an ideal, although limited size (num qubits) ideal device. As in is the only reason I am doing this that a quantum computer can't handle it - but I imagine with less features it may be easier/quicker to train.

Yes, the same concerns. The idea is to reduce the problem to a smaller one. But at a cost of potentially other problems. Yes, you may train a model quicker, but performance may be worse. This is exactly what we see in the PCA section.